great_tables/_formats.py

```diff
@@ -285,66 +286,129 @@ def fmt_number(
     sep_mark = _get_locale_sep_mark(default=sep_mark, use_seps=use_seps, locale=locale)
     dec_mark = _get_locale_dec_mark(default=dec_mark, locale=locale)
 
-    # Generate a function that will operate on single `x` values in the table body
```

Comment:
The `fmt_number_fn()` function is extracted from the body of the `fmt_number()` function.

This is a general pattern that will be used throughout this file. The extracted function is migrated to a new function that has a `context=` argument (for `"html"` and `"latex"` contexts). This function will be used in a `partial()` statement for each of the supported contexts (all within a `fmt()` call).